include/cxx.h

```diff
@@ -297,6 +297,93 @@ class Box final {
 };
 #endif // CXXBRIDGE1_RUST_BOX
 
+#ifndef CXXBRIDGE1_RUST_OPTION
+template <typename T>
+class Option final {};
+
+template <typename T>
+class Option<T&> {
+public:
+  Option() noexcept;
+  Option(Option&&) noexcept;
+  Option(T&) noexcept;
+  ~Option() noexcept;
+
+  Option<T&>& operator=(Option&&) noexcept;
+  const T& operator*() const noexcept;
+  const T* operator->() const noexcept;
+  T& operator*() noexcept;
+  T* operator->() noexcept;
```

Comment:
@capickett could you double check the type signatures for operator* and operator->? I vaguely remember there being some invariant you have to uphold about the relationship between the two methods, but I couldn't find anything straightforward.

Reply:
Looks right to me. I referenced https://en.cppreference.com/w/cpp/language/operator_member_access